lib/services/ngsi/subscriptionService.js

```diff
@@ -0,0 +1,211 @@
+                }
+            ],
+            reference: config.providerUrl + '/notify',
+            duration: config.deviceRegistrationDuration || 'P1M',
+            notifyConditions: [
+                {
+                    type: 'ONCHANGE',
+                    condValues: triggers
+                }
+            ],
+            throttling: config.throttling || 'PT5S'
+        }
+    };
+
+    if (device.cbHost) {
+        options.uri = 'http://' + device.cbHost + '/v1/subscribeContext';
+    } else {
+        options.uri = 'http://' + config.contextBroker.host + ':' + config.contextBroker.port + '/v1/subscribeContext';
+    }
+
+    request(options, createSubscriptionHandler(device, triggers, callback));
+}
+
+/**
+ * Generate a new unsubscription request handler, based on the device and subscription ID given to the function.
+ *
+ * @param {Object} device       Object containing all the information about a particular device.
+ * @param {String} id           ID of the subscription to remove.
+ * @param {Function} callback   Callback to be called when the subscription handler ends.
+ * @return {Function}           Returns a request handler for the given data.
+ */
+function createUnsuscribeHandler(device, id, callback) {
+    return function(error, response, body) {
+        if (error || !body) {
+            logger.debug(
+                context,
+                'Transport error found subscribing device with id [%s] to entity [%s]', device.id, device.name);
+
+            callback(error);
+
+        } else if (response.statusCode !== 200 && response.statusCode !== 201) {
+            logger.debug(context, 'Unknown error subscribing device with id [%s] to entity [%s]: $s',
+                response.statusCode);
+
+            callback(new errors.EntityGenericError(device.name, device.type, {
+                details: body
+            }, response.statusCode));
+
+        } else if (body && body.orionError) {
+            logger.debug(
+                context,
+                'Orion found subscribing device with id [%s] to entity [%s]: %s',
+                device.id, device.name, body.orionError);
+
+            callback(new errors.BadRequest(body.orionError.details));
+        } else {
+            device.subscriptions.splice(device.subscriptions.indexOf(id), 1);
+            deviceRegistry.update(device, callback);
+        }
+    };
+}
+
+/**
+ * Remove the subscription with the given ID from the Context Broker and from the device repository.
+ *
+ * @param {Object} device       Object containing all the information about a particular device.
+ * @param {String} id           ID of the subscription to remove.
+ */
+function unsubscribe(device, id, callback) {
+    var options = {
+        method: 'POST',
+        headers: {
+            'fiware-service': device.service,
+            'fiware-servicepath': device.subservice
+        },
+        json: {
+            subscriptionId: id,
+            duration: 'P0S'
+        }
+
+    };
```

Comment:
Wouldn't it be better to use either POST /v1/unsubscribeContext or convop DELETE /v1/contextSubscriptions/{subscriptionID}? I think both of them remove the subscription form orion DB. (At least the convop does)